ds-collector/ds-collector

```diff
@@ -722,6 +705,81 @@ collect_dse_solr_cores() {
     fi
 }
 
+discover_jmx() {
+ tmp=""
+
+  # XXX from Cassandra 4.0 it is possible to have multiple process per ip, how can we handle this?
+  server_cmdline="$(ps -aef|grep -e 'org.apache.cassandra.service.CassandraDaemon' -e 'com.datastax.bdp.DseModule'|grep java)"
+  tmp=$(echo "${server_cmdline}" | grep 'cassandra.jmx.local.port=' |sed -e 's|^.*-Dcassandra.jmx.local.port=\([^ ]*\).*$|\1|')
+  if [ -n "$tmp" ]; then
+      jmxPort="$tmp"
+  else
+      tmp=$(echo "${server_cmdline}" | grep 'cassandra.jmx.remote.port=' |sed -e 's|^.*-Dcassandra.jmx.remote.port=\([^ ]*\).*$|\1|')
+      if [ -n "$tmp" ]; then
+          jmxPort="$tmp"
+      fi
+      tmp=$(echo "${server_cmdline}" | grep 'java.rmi.server.hostname=' |sed -e 's|^.*-Djava.rmi.server.hostname=\([^ ]*\).*$|\1|')
+      if [ -n "$tmp" ]; then
+          jmxHost="$tmp"
+      fi
+  fi
+  if [ -n "$(command -v nc)" ]; then
+    # test jmxHost, failling back if need be
+    $timeout_command  nc -zv "${jmxHost}" "${jmxPort}" > /dev/null 2>&1
+    if [ $? != 0 ]; then
+      $timeout_command nc -zv localhost $jmxPort > /dev/null 2>&1
+      if [ $? = 0 ]; then
+        jmxHost='localhost'
+      else
+        $timeout_command nc -zv $(hostname) $jmxPort > /dev/null 2>&1
+        if [ $? = 0 ]; then
+          jmxHost="$(hostname)"
+        fi
+      fi
+    fi
+  else
+    echo "nc not found, auto-test of jmxHost disabled"
+  fi
+  $timeout_command  nc -zv "${jmxHost}" "${jmxPort}" > /dev/null 2>&1
```

Comment:
We just did a test to see that command existed above with an `if...else` block, should this section not move up there?